spec.html

```diff
@@ -37807,23 +37818,48 @@ <h1>AtomicLoad ( _typedArray_, _index_ )</h1>
+        <emu-alg>
+          1. Assert: _xBytes_ and _yBytes_ have equal length.
+          1. Let _result_ be a new empty List.
+          1. Let _i_ be 0.
+          1. Let each element _xByte_ of _xBytes_, do
+            1. Let _yByte_ be _yBytes_[_i_].
+            1. Let _resultByte_ be the result of applying the bitwise operator _op_ to _xByte_ and _yByte_.
+            1. Set _i_ to _i_ + 1.
+            1. Append _resultByte_ to the end of _result_.
+          1. Return _result_.
+        </emu-alg>
+      </emu-clause>
     </emu-clause>
 
     <emu-clause id="sec-atomics.add">
       <h1>Atomics.add ( _typedArray_, _index_, _value_ )</h1>
-      <p>Let `add` denote a semantic function of two List of byte values arguments that applies the addition operation to the Number values corresponding to the List of byte values arguments and returns a List of byte values corresponding to the result of that operation.</p>
       <p>The following steps are taken:</p>
       <emu-alg>
-        1. Return ? AtomicReadModifyWrite(_typedArray_, _index_, _value_, `add`).
+        1. Let _type_ be the Element Type value in <emu-xref href="#table-the-typedarray-constructors"></emu-xref> for _typedArray_.[[TypedArrayName]].
+        1. Let _isLittleEndian_ be the value of the [[LittleEndian]] field of the surrounding agent's Agent Record.
+        1. Let _add_ be a new read-modify-write modification function with parameters (_xBytes_, _yBytes_) that captures _type_ and _isLittleEndian_ and performs the following steps atomically when called:
+          1. Assert: _xBytes_ and _yBytes_ have equal length.
+          1. Let _x_ be RawBytesToNumeric(_type_, _xBytes_, _isLittleEndian_).
+          1. Let _y_ be RawBytesToNumeric(_type_, _yBytes_, _isLittleEndian_).
+          1. Let _T_ be Type(_x_).
+          1. Let _sum_ be _T_::add(_x_, _y_).
+          1. Return NumericToRawBytes(_type_, _sumBytes_, _isLittleEndian_).
+        1. Return ? AtomicReadModifyWrite(_typedArray_, _index_, _value_, _add_).
       </emu-alg>
     </emu-clause>
 
     <emu-clause id="sec-atomics.and">
       <h1>Atomics.and ( _typedArray_, _index_, _value_ )</h1>
-      <p>Let `and` denote a semantic function of two List of byte values arguments that applies the bitwise-and operation element-wise to the two arguments and returns a List of byte values corresponding to the result of that operation.</p>
       <p>The following steps are taken:</p>
       <emu-alg>
-        1. Return ? AtomicReadModifyWrite(_typedArray_, _index_, _value_, `and`).
+        1. Let _and_ be a new read-modify-write modification function with parameters (_xBytes_, _yBytes_) that captures nothing and performs the following steps atomically when called:
+          1. Return ByteListBitwiseOp(`&`, _xBytes_, _yBytes_).
```

Comment:
```suggestion
          1. Return ! ByteListBitwiseOp(`&`, _xBytes_, _yBytes_).
```

Reply:
See above for not returning completion values or call other non-completion value returning ops so as to appear obviously pure. Lack of ?/! is intentional here.